### PR TITLE
now dead dog, cannots lick u

### PR DESCRIPTION
**Pull request by AnywayFarus**
## About The Pull Request
Dead dog cannot befrind of you, and lick your hand

## Why It's Good For The Game

Add logic.. or realistic.

## Changelog

:cl: Vishenka0704
fix: Dead dogs can no longer befriend you or lick you.
/:cl:

### CODE REVIEW

File: code/datums/ai/dog/dog_controller.dm [hunk 1]
```diff
@@ -97,6 +97,10 @@
 /datum/ai_controller/dog/proc/on_attack_hand(datum/source, mob/living/user)
 	SIGNAL_HANDLER
 
+	var/mob/living/living_pawn = pawn
+	if(living_pawn.stat == DEAD)
+		to_chat(user, span_warning("When you touch [pawn], you feel a dead cold from their body."))
```
Comment (san7890) [inline code/datums/ai/dog/dog_controller.dm:102]: This reads kinda awkwardly, can you rephrase?

Comment (san7890) [reply code/datums/ai/dog/dog_controller.dm:102]: ALSO, we have that not-so-secret feature in the codebase where clowns don't understand death. Might be nice to have a check for that :)

Comment (AnywayFarus) [reply code/datums/ai/dog/dog_controller.dm:102]: it may be worth deleting this message in the chat, I doubt the need for this.

**Comment by tralezab** [comment]
dog ai :(((((((((((((((((((((((((((((

**Comment by AnywayFarus** [comment]
I had to kill corgis to test it. not cool

File: code/datums/ai/dog/dog_controller.dm [hunk 2]
```diff
@@ -97,6 +97,12 @@
 /datum/ai_controller/dog/proc/on_attack_hand(datum/source, mob/living/user)
 	SIGNAL_HANDLER
 
+	var/mob/living/living_pawn = pawn
+	var/pronounce = pawn.gender == "male" ? "his" : "her"
+	var/additional_text = HAS_TRAIT(user, TRAIT_NAIVE) ? "Maybe bring a heater?" : "It looks like a corpse."
+	if(living_pawn.stat == DEAD)
+		to_chat(user, span_warning("Touching [pawn], you feel [pronounce] cold skin through the fur. [additional_text]"))
+		return
```
Comment (timothymtorres) [inline code/datums/ai/dog/dog_controller.dm:105]:
```suggestion
	var/additional_text = HAS_TRAIT(user, TRAIT_NAIVE) ? "It looks like [living_pawn.p_they()] are sleeping." : "It looks like a corpse."
	if(living_pawn.stat == DEAD)
		to_chat(user, span_warning("Touching [living_pawn], you feel [living_pawn.p_their()] cold skin through the fur. [additional_text]"))
		return
```

I'd suggest you look at https://github.com/tgstation/tgstation/blob/master/code/__HELPERS/pronouns.dm since these handle grammar stuff.

Comment (AnywayFarus) [reply code/datums/ai/dog/dog_controller.dm:105]: oh. What is the connection between cold skin and sleep?
Maybe leave the option with a heater?

Comment (san7890) [reply code/datums/ai/dog/dog_controller.dm:105]: Mammals naturally drop in body temperature when they go to sleep, so it's not an illogical conclusion to make. I think the heater thing is cute though.

The key point about this suggestion is using `p_they` and `p_their` here, because we already naturally track pronouns on all mobs, you don't need to add the

```dm
var/pronounce = pawn.gender == "male" ? "his" : "her"
```

check in code. Much cleaner to do it as tim does it above.

**Comment by Imaginos16** [comment]
> 🆑 Vishenka0704 fix: dead dog cannots lick you and befriend with you /🆑

Really hate to be that guy, but you should edit your CL to say "Dead dogs can no longer befriend you or lick you."

**Comment by AnywayFarus** [comment]
> > 🆑 Vishenka0704 fix: dead dog cannots lick you and befriend with you /🆑
> 
> Really hate to be that guy, but you should edit your CL to say "Dead dogs can no longer befriend you or lick you."

Quite well marked. Correcting grammatical errors in a friendly manner is great.

File: code/datums/ai/dog/dog_controller.dm [hunk 3]
```diff
@@ -97,6 +97,11 @@
 /datum/ai_controller/dog/proc/on_attack_hand(datum/source, mob/living/user)
 	SIGNAL_HANDLER
 
+	var/mob/living/living_pawn = pawn
+	var/additional_text = HAS_TRAIT(user, TRAIT_NAIVE) ? "It looks like [living_pawn.p_they()] are sleeping." : "It looks like a corpse."
```
Comment (dragomagol) [inline code/datums/ai/dog/dog_controller.dm:101]:
```suggestion
	var/additional_text = HAS_TRAIT(user, TRAIT_NAIVE) ? "It looks like [living_pawn.p_they()] [living_pawn.p_are()] sleeping." : "Seems to be dead."
```

Comment (dragomagol) [reply code/datums/ai/dog/dog_controller.dm:101]: This code should also be inside the if block

**Comment by Imaginos16** [comment]
> > > 🆑 Vishenka0704 fix: dead dog cannots lick you and befriend with you /🆑
> > 
> > 
> > Really hate to be that guy, but you should edit your CL to say "Dead dogs can no longer befriend you or lick you."
> 
> Quite well marked. Correcting grammatical errors in a friendly manner is great.

No worries! Just making sure CL looks fine!

File: code/datums/ai/dog/dog_controller.dm [hunk 4]
```diff
@@ -97,6 +97,11 @@
 /datum/ai_controller/dog/proc/on_attack_hand(datum/source, mob/living/user)
 	SIGNAL_HANDLER
 
+	var/mob/living/living_pawn = pawn
+	var/additional_text = HAS_TRAIT(user, TRAIT_NAIVE) ? "It looks like [living_pawn.p_they()] [living_pawn.p_are()] sleeping." : "[living_pawn.p_they(capitalized = TRUE)] seem[living_pawn.p_s()] to be dead."
```
Comment (ShizCalev) [inline code/datums/ai/dog/dog_controller.dm:101]:
```suggestion
	var/additional_text = HAS_TRAIT(user, TRAIT_NAIVE) ? "It looks like [living_pawn.p_theyre()] sleeping." : "[living_pawn.p_they(capitalized = TRUE)] seem[living_pawn.p_s()] to be dead."
```

Comment (AnywayFarus) [reply code/datums/ai/dog/dog_controller.dm]: funny haha
